src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -2310,6 +2310,47 @@ public sealed class SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http2 :
     {
         public SocketsHttpHandlerTest_HttpClientHandlerTest_Headers_Http2(ITestOutputHelper output) : base(output) { }
         protected override Version UseVersion => HttpVersion.Version20;
+
+        [Fact]
+        public async Task SendAsync_LargeHeaders_Continuation_CorrectlyWritten()
```

Comment:
I think this test should be put in HttpClientHandlerTest.Http2.cs. That's where most of the HTTP2 specific tests are.

I also think that this test should specifically check that the HEADERS and CONTINUATION frames are sent as we expect them to be (i.e. one HEADER frame of max size, then a single CONTINUATION with the rest of the header data). The HTTP2 tests in HttpClientHandlerTest.Http2.cs generally try to validate the specific frames being sent, instead of just using the generic loopback logic. That's intentional; these tests are HTTP2 specific and we want to validate the specific HTTP2 low-level behavior.

I also think we should consider adding another test that causes a second CONTINUATION to be sent, so that it's HEADERS, CONTINUATION w/o EndHeaders, CONTINUATION with EndHeaders.

Reply:
Moved and extended to test for 0, 1 and 2 CONTINUATION frames in a request, server side reads the frames and checks the number of continuations.